src/sagas/pushNotification.js

```diff
@@ -0,0 +1,432 @@
+        return '';
+      }
+      const countArgs = args.length;
+      if (countArgs === 3) {
+        const [firstToken, secondToken, other] = args;
+        const otherCount = parseInt(other, 10);
+        message = t`You have received ${firstToken}, ${secondToken} and ${otherCount} other token on a new transaction.`;
+      } else if (countArgs === 2) {
+        const [firstToken, secondToken] = args;
+        message = t`You have received ${firstToken} and ${secondToken}.`;
+      } else if (countArgs === 1) {
+        const [firstToken] = args;
+        message = t`You have received ${firstToken}.`;
+      }
+    } else if (key === NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS) {
+      message = t`There is a new transaction in your wallet.`;
+    } else if (key === NEW_TRANSACTION_RECEIVED_TITLE) {
+      message = t`New transaction received`;
+    }
+    return message;
+  }
+};
+
+const onForegroundMessage = async (message) => {
+  await messageHandler(message);
+};
+
+const onBackgroundMessage = async (message) => {
+  await messageHandler(message);
+};
+
+/**
+ * Handle the message received when application is in foreground and background (not closed) state
+ * @param {{ data: Object, from: string, messageId: string, sentTime: number, ttl: number }} message - Message received from wallet-service
+ * @example
+ * {
+ *   bodyLocArgs: '[\"10 T2\",\"5 T1\",\"2\"]',
+ *   bodyLocKey: 'new_transaction_received_description_with_tokens',
+ *   titleLocKey: 'new_transaction_received_title',
+ *   txId: 'txId1',
+ * }
+ * @inner
+ */
+const messageHandler = async (message) => {
+  const { data } = message;
+  if (!localization.hasKey(data.titleLocKey)) {
+    console.log('unknown message titleLocKey', data.titleLocKey);
+    return;
+  }
+
+  if (!localization.hasKey(data.bodyLocKey)) {
+    console.log('unknown message bodyLocKey', data.bodyLocKey);
+    return;
+  }
+
+  const bodyArgs = JSON.parse(data.bodyLocArgs);
+  const title = localization.getMessage(data.titleLocKey);
+  const body = localization.getMessage(data.bodyLocKey, bodyArgs);
+
+  notifee.displayNotification({
+    title,
+    body,
+    android: {
+      channelId: TRANSACTION_CHANNEL_ID,
+    },
+  });
+};
+
+const showPinScreenForResult = async (dispatch) => new Promise((resolve) => {
+  const params = {
+    cb: (_pin) => {
+      dispatch(setIsShowingPinScreen(false));
+      resolve(_pin);
+    },
+    canCancel: false,
+    screenText: t`Enter your 6-digit pin to authorize operation`,
+    biometryText: t`Authorize operation`,
+  };
+
+  NavigationService.navigate('PinScreen', params);
+
+  // We should set the global isShowingPinScreen
+  dispatch(setIsShowingPinScreen(true));
+});
+
+/**
+ * This function is called when the wallet is initialized with success.
+ */
+export function* onAppInitialization() {
+  yield take(types.START_WALLET_SUCCESS);
+
+  const { enabled, showAmountEnabled } = STORE.getItem(pushNotificationKey.settings);
```

Comment:
Will you have this in the STORE the first time you start the wallet with this new feature? Or `STORE.getItem(pushNotificationKey.settings)` will return `null`?

Reply:
Good point. I can add a fallback. But it is working anyway.